frontend/javascripts/admin/dataset/composition_wizard/03_select_datasets.tsx

```diff
@@ -0,0 +1,70 @@
+import { Button } from "antd";
+import { AsyncButton } from "components/async_clickables";
+import DatasetSelectionComponent, {
+  DatasetSelectionValue,
+} from "dashboard/dataset/dataset_selection_component";
+import React, { useEffect, useState } from "react";
+import { tryToFetchDatasetsByName, WizardComponentProps } from "./common";
+
+export default function SelectDatasets({ wizardContext, setWizardContext }: WizardComponentProps) {
+  const [datasetValues, setDatasetValues] = useState<DatasetSelectionValue[]>([]);
+
+  const onPrev = () => {
+    setWizardContext((oldContext) => ({
+      ...oldContext,
+      currentWizardStep:
+        wizardContext.composeMode === "WITHOUT_TRANSFORMS" ? "SelectImportType" : "UploadFiles",
+    }));
+  };
+  const onNext = async () => {
+    const datasets = await tryToFetchDatasetsByName(
+      datasetValues.map((el) => el.value),
+      "Could not find datasets. Please doublecheck your selection.",
+    );
+    if (datasets == null) {
+      // An error message was already shown in tryToFetchDatasetsByName
+      return;
+    }
+
+    setWizardContext((oldContext) => ({
+      ...oldContext,
+      currentWizardStep: "ConfigureNewDataset",
+      datasets,
+    }));
+  };
+
+  useEffect(() => {
+    setDatasetValues(wizardContext.datasets.map((ds) => ({ value: ds.name, label: ds.name })));
+  }, []);
+
+  // When not using any transforms,
+  let isDatasetCountValid = true;
+  if (wizardContext.composeMode === "WITHOUT_TRANSFORMS") {
+    isDatasetCountValid = datasetValues.length > 0;
+  } else {
+    isDatasetCountValid = datasetValues.length === 2;
```

Comment:
Do you think a use case could also be to transform two layers of the same dataset using BigWarp or wk NMLs?

Reply:
I think, this usecase is imaginable, but not the default one. One could argue that two layers should only be put into the same DS if they are aligned with each other. If they still need alignment, they should be put into separate datasets and then be combined using this new workflow. Of course, it will happen that this is not the case, but I wouldn't want to design for this (at least not in this iteration).